### DISCUSSION TITLE
How to Disable Smooth Transitions for Zooming?

### DISCUSSION THREAD

**Discussion by mikisf** [yomotsu/camera-controls#551] [Q&A] [answered]
Hi,

I’m using camera-controls in my Three.js project, and I want to completely disable smooth transitions for camera movements (orbiting, panning, and zooming). Specifically, I’d like all movements to happen instantly without easing or damping.

Here’s what I’ve tried so far:

```
cameraControls.smoothTime = 0;
cameraControls.draggingSmoothTime = 0;
```
This removes some smoothing for orbiting and panning, but it doesn't affect zooming. Is there a specific way to configure camera-controls for snappy, instant camera movements for all interactions (orbit, pan, zoom)?

Any guidance or clarification would be greatly appreciated!

Thanks for the awesome library! 🙌

**Comment by yomotsu** [comment, marked as answer]
sorry for the delay,

The smooth behavior in `_zoomInternal` cannot be directly modified:
https://github.com/yomotsu/camera-controls/blob/582c9be6b76e0f1850cc55eefa9e925ea5dae0ff/src/CameraControls.ts#L3125

However, you can customize the internal method by extending the CameraControls class.
Here's an example:
```
class MyCameraControls extends CameraControls {

  protected _zoomInternal = ( delta: number, x: number, y: number ): void => {

		const zoomScale = Math.pow( 0.95, delta * this.dollySpeed );
		const lastZoom = this._zoom;
		const zoom = this._zoom * zoomScale;

		this.zoomTo( zoom, false ); // change here

		if ( this.dollyToCursor ) {

			this._changedZoom += zoom - lastZoom;
			this._dollyControlCoord.set( x, y );

		}

	};

}
```
You can then integrate `MyCameraControls` into your project as needed.

**Reply by mikisf** [reply]
Thank you for your detailed explanation! This is exactly what I needed. I'll adapt it to my project—really appreciate your help! 😊
